### ISSUE TITLE
[BUG] Parquet column selection by name with schemas including   list<struct<X, Y>> does not work.

### ISSUE THREAD

**Issue by nvdbaranec** [rapidsai/cudf#14539]
If you have a schema that contains a list-of-struct, selecting a subset of the inner columns doesn't work.  Example

`list<struct<int, float>>`
If the schema for this column was

```
A           (list)
   B        (struct)
       C    (int)
       D    (float)
```
Attempting to select "A.B.C" would not work.  I believe this is being caused by some schema preprocessing that we are doing that is injecting fake schema elements to ease schema interpretation.  Essentially we see a schema that looks like this:

```
A           (list)
  list      (the fake element
   B        (struct)
       C    (int)
       D    (float)
```
So "A.B.C" doesn't actually exist, only "A.list.B.C" and the code returns 0 columns.

**Comment by nvdbaranec** [comment]
Actually, upon further review, this mystery "list" element is in the parquet file itself (it's one of the odd ways in which the spec allows you to specify list columns).  A question here though would be what would a user expect to be the correct way to do it.  For Pandas or Spark, would you expect to have to put "list" in there when selecting a subset of columns?  @jlowe  @shwina

**Comment by hyperbolic2346** [comment]
The schema for this part of the file is 
```
message spark_schema {
  required int32 id;
  optional group phoneNumbers {
    repeated group phone {
      required int64 number;
      optional binary kind (STRING);
    }
  }
}
```

**Comment by nvdbaranec** [comment]
Right, that's the question:  are these details something you'd expect the end user to know or care about, or would they just expect "A.B.C"?  Maybe this is a what-would-Pandas-do question.

**Comment by etseidl** [comment]
@nvdbaranec It's been a few years, but I believe the way to query in the above situation is to use `explode` to convert the list to separate rows.  If there were another column at the top of the hierarchy ('X'), then the value for 'X' would be repeated for each new row that the list 'A' was exploded into. Here's a pyspark query I did years ago against the data @hyperbolic2346 quoted above:
```python
df.createOrReplaceTempView("asm")

sql = """
select func_name, func_addr_start, blk_addr_start, blk_id, flatten(sources.asm) as asm from (
  select func_name, func_addr_start, bb.blk_addr_start, bb.blk_id, filter(bb.sources,x->x.asm_scrub_type = 'no_scrub') as sources
    from (select func_name, func_addr_start, explode(basic_blocks) as bb from asm))
where func_name='introduce'
""" 
```
